src/libraries/System.Private.CoreLib/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -21,8 +22,9 @@ internal enum BufferAllocatedReason : int
             PoolExhausted
         }
 
-        // The ArrayPoolEventSource GUID is {0866b2b8-5cef-5db9-2612-0c0ffd814a44}
-        private ArrayPoolEventSource() : base(new Guid(0x0866b2b8, 0x5cef, 0x5db9, 0x26, 0x12, 0x0c, 0x0f, 0xfd, 0x81, 0x4a, 0x44), "System.Buffers.ArrayPoolEventSource") { }
+        // Parameterized constructor to block initialization and ensure the EventSourceGenerator is creating the default constructor
+        // as you can't make a constructor partial.
+        private ArrayPoolEventSource(int _) { }
```

Comment:
@jaredpar is there a canonical way to do this with source generators?  I'm not opposed to this method, though.

Reply:
`private partial ArrayPoolEventSource();` would have worked if it was allowed (though isn't, only works for regular methods, not constructors)

The main aim is to fail compilation if the generator didn't run; rather than to succeed but not use the generated paths.